core/src/test/java/kafka/test/MockController.java

```diff
@@ -77,8 +82,28 @@ private MockController(Collection<MockTopic> initialTopics) {
     }
 
     @Override
-    public CompletableFuture<CreateTopicsResponseData> createTopics(CreateTopicsRequestData request) {
-        throw new UnsupportedOperationException();
+    synchronized public CompletableFuture<CreateTopicsResponseData>
+            createTopics(CreateTopicsRequestData request) {
+        CreateTopicsResponseData response = new CreateTopicsResponseData();
+        for (CreatableTopic topic : request.topics()) {
+            if (topics.containsKey(topic.name())) {
+                response.topics().add(new CreatableTopicResult().
+                    setName(topic.name()).
+                    setErrorCode(Errors.TOPIC_ALREADY_EXISTS.code()));
+            } else {
+                long topicId = nextTopicId.getAndAdd(1);
+                Uuid topicUuid = new Uuid(0, topicId);
+                topicNameToId.put(topic.name(), topicUuid);
+                topics.put(topicUuid, new MockTopic(topic.name(), topicUuid));
+                response.topics().add(new CreatableTopicResult().
+                    setName(topic.name()).
+                    setErrorCode(Errors.NONE.code()).
+                    setTopicId(topicUuid));
+                // For a better mock, we might want to return configs, replication
```

Comment:
agreed. why not including such information?

Reply:
It wouldn't be useful for the test in this PR since we're trying to test the logic in `ControllerApis.scala`, which doesn't interact with configs or the replication factor.